### ISSUE TITLE
Remove custom reconnect code

### ISSUE THREAD

**Issue by SimonWoolf** [ably/demo-mobile-ios#6]
So a customer today was saying they thought they had to implement reconnecting on disconnect / suspended themselves, after seeing the following in this demo:

```
realtime.connection.on { stateChange in
    if let stateChange = stateChange {
        self.delegate?.chatModel(self, connectionStateChanged: stateChange)

        switch stateChange.current {
        case .Disconnected:
            self.attemptReconnect(5000)
        case .Suspended:
            self.attemptReconnect(15000)
        default:
            break
        }
    }
}
```

Given this is a demo app, re-implementing functionality that's already in the client library is this is going to mislead customers into thinking they need to do the same thing, when they don't.

If it's important that the timeouts be reduced from 30/15 to 15/5, then this those options should be passed in in the clientOptions. But really, I think we should leave them at the defaults, the demo app should be as simple as possible -- if we think the defaults are so wrong for everyone that we want to override them in the demo app, we should just change the defaults.

@stannedelchev, cc @mattheworiordan

**Comment by mattheworiordan** [comment]
@SimonWoolf agreed, this was legacy code from before the options to change the defaults was available and also from before iOS detected OS network events and automatically reconnected.  I agree we should just remove completely from the demo
